pypeit/core/quicklook.py

```diff
@@ -0,0 +1,312 @@
+
+    # Rename calibs
+    calib_pypeit_files = []
+    for pypeit_file, setup in zip(pypeit_files, setups):
+
+        # Rename with _calib
+        calib_pypeit_file = pypeit_file.replace('_{}.pypeit'.format(setup),
+                                                '_calib_{}.pypeit'.format(setup))
+        os.rename(pypeit_file, calib_pypeit_file)
+        calib_pypeit_files.append(calib_pypeit_file)
+
+    return calib_pypeit_files
+
+def process_calibs(calib_pypeit_files:list):
+    """Process the calibrations
+
+    Args:
+        calib_pypeit_files (list): 
+            List of PypeIt files for the calibrations
+    """
+
+    # Loop on setups, rename + run calibs
+    for calib_pypeit_file in calib_pypeit_files: 
+        # Run me via the script
+        redux_path = os.path.dirname(calib_pypeit_file)  # Path to PypeIt file
+        run_pargs = run_pypeit.RunPypeIt.parse_args(
+            [calib_pypeit_file, 
+             '-r={}'.format(redux_path), '-c'])
+        run_pypeit.RunPypeIt.main(run_pargs)
+
+
+def folder_name_from_scifiles(sci_files:list):
+    """ Folder name for output of QL on science file(s)
+
+    Currently, the code takes the name of the first file.
+    This may evolve.. 
+
+    Args:
+        sci_files (list): List of science files
+
+    Returns:
+        str: Folder name
+    """
+    # For now, we return the first filename
+    #  without .fits
+    return os.path.splitext(os.path.basename(sci_files[0]))[0]
+
+def generate_sci_pypeitfile(calib_pypeit_file:str, 
+                            redux_path:str, 
+                            sci_files:list, 
+                            ps_sci_list:list, 
+                            input_cfg_dict:dict={}, 
+                            remove_sci_dir:bool=True, 
+                            maskID:str=None):
+    """
+    Generate the PypeIt file for the science frames
+    from the calib PypeIt file
+    
+    Args:
+        calib_pypeit_file (str): Calibration PypeIt file
+        redux_path (str): Path to the redux folder
+        sci_files (list): List of science files
+        ps_sci_list (list): List of pypeit.setup.PypeItSetup objects
+        input_cfg_dict (dict, optional): Input configuration dictionary. Defaults to {}.
+        remove_sci_dir (bool, optional): Remove the science directory if it exists. Defaults to True.
+        maskID (str, optional): Mask ID to isolate for QL.  Defaults to None.
+
+    Returns: 
+        tuple: science_pypeit (str), pypeitFile (pypeit.inputfiles.PypeItFile)
+    """
+
+    # Parse science file info
+    #science_file = os.path.join(pargs.full_rawpath, pargs.science)
+    science_pypeit = calib_pypeit_file.replace('calib', 'science')
+
+    folder = folder_name_from_scifiles(sci_files)
+    sci_dir = os.path.join(redux_path, folder)
+    master_dir = os.path.join(sci_dir, 'Masters')
+
+    # Science reuction folder
+    if os.path.isdir(sci_dir) and remove_sci_dir:
+        os.system('rm -rf {}'.format(sci_dir))
```

Comment:
I had a really strong (over)reaction to this.  It feels dangerous, even if the alternative is basically the same.  Can we use `shutil.rmtree` instead?

Reply:
I agree on this one, `shutil.rmtree` feels safer. Even though that's probably not actually true...

Reply:
can do, and have.